src/odoc.ml

```diff
@@ -129,17 +129,21 @@ let module_deps (m : Module.t) ~doc_dir ~(dep_graphs:Dep_graph.Ml_kind.t) =
    else
      (* When a module has no .mli, use the dependencies for the .ml *)
      Dep_graph.deps_of dep_graphs.impl m)
-  >>^ List.map ~f:(Module.odoc_file ~doc_dir)
+  >>^ List.map ~f:(fun m -> Path.build (Module.odoc_file ~doc_dir m))
   |> Build.dyn_paths
 
 let compile_module sctx (m : Module.t) ~includes:(file_deps, iflags)
       ~dep_graphs ~doc_dir ~pkg_or_lnu =
-  let odoc_file = Module.odoc_file m ~doc_dir in
+  let odoc_file =
+    Module.odoc_file m ~doc_dir
+    |> Path.build
```

Comment:
Why `|>` rather than `Path.build (Module.odoc_file m ~doc_dir)`? `|>` makes me thing of piping data transformations. Wo as a rule of thumb, this is how I choose between `x |> f |> g` and `g (f x)`:

- if `g` and `f` just enclose their input into a bigger data structure, I use `g (f x)`, i.e. we just construct a value
- if `g` and `f` transforms their input, I use `x |> f |> g` to show the various transformations steps

Reply:
I like `|>` because it's easier to refactor. You can just re-order lines around rather than deal with matching parens. I switched back to normal function application.